### PR TITLE
For terminal goat232's trashy code

### PR DESCRIPTION
**Pull request by TerminalGoat232**

### CODE REVIEW

File: script/mouse_controller.gd [hunk 1]
```diff
@@ -25,15 +25,12 @@ func _unhandled_input(event: InputEvent) -> void:
...
 			MOUSE_BUTTON_WHEEL_UP:
-
 				target_pos -= %camera.global_basis.z * 2
 #desperation
-func vpclamp(v: Vector3,v_min: Vector3,v_max: Vector3, offset: float = 0) -> Vector3:
-	v.x = clamp(v.x, v_min.x - offset, v_max.x + offset)
-	v.y = clamp(v.y, v_min.y - offset, v_max.y + offset)
-	v.z = clamp(v.z, v_min.z - offset, v_max.z + offset)
-	return Vector3(v.x, v.y, v.z)
+func boxed_clamp(u: Vector3, v_m: Vector3, offset: float = 0) -> Vector3:
```
Comment (mncc8337) [inline script/mouse_controller.gd]: ddooir v_m thanhf box_origin dc k

**Review by mncc8337 — CHANGES_REQUESTED** [review]
sap xong r

File: script/mouse_controller.gd (hunk 1 above)
Comment (TerminalGoat232) [reply script/mouse_controller.gd]: ok

**Comment by mncc8337** [comment]
ee maf nhows laf cais %platform nows laf tamj thowif thooi nghe. vai buwax laf hn thanhf cais khacs neen t ms keeu laf neen laays %camera.mousecast()

File: script/mouse_controller.gd [hunk 2]
```diff
@@ -27,3 +31,11 @@ func _unhandled_input(event: InputEvent) -> void:
 				target_pos += %camera.global_basis.z * 2
 			MOUSE_BUTTON_WHEEL_UP:
 				target_pos -= %camera.global_basis.z * 2
+
+#desperation
+func boxed_clamp(u: Vector3, box_center: Vector3, offset: float = 0) -> Vector3:
```
Comment (mncc8337) [inline script/mouse_controller.gd:36]: ok

**Review by mncc8337 — APPROVED** [review]

**Pull request merged**